ipc_sendfile.c

```diff
@@ -340,19 +348,80 @@ int ipc_send_pipe(char* fileName){
 	return 0;
 }
 
+int ipc_send_mqueue(char* fileName){
+	struct mq_attr attrs;
+	mqd_t msg_queue;
+	int status;
+	char * buffer;
+
+
+	int part_nb;
+
+	int fileSize;
+
+	printf("Creating Queue... \n");
+
+	memset(&attrs, 0, sizeof attrs);
+	attrs.mq_maxmsg = MAX_MSG_QUEUE;
+	attrs.mq_msgsize = MAX_MSG_QUEUE_SIZE;
+	msg_queue = mq_open( "queue", O_WRONLY | O_CREAT, S_IRWXU | S_IRWXG, &attrs );
+	if (msg_queue == -1) {
+		perror ("mq_open()");
+	    return EXIT_FAILURE;
+	}
```

Comment:
if the queue already exist and have some message on it, you won't see it and the receivefile will get them first -> the copy won't be exact.
You should unlink the queue first or use O_EXCL to know if the queue already exists.

Reply:
Fixed!